apiml-common/src/main/java/com/ca/mfaas/product/gateway/GatewayInstanceInitializer.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package com.ca.mfaas.product.gateway;
+
+import com.ca.mfaas.product.constants.CoreService;
+import com.ca.mfaas.product.instance.InstanceInitializationException;
+import com.ca.mfaas.product.instance.lookup.InstanceLookupExecutor;
+import com.netflix.appinfo.InstanceInfo;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.event.EventListener;
+
+import java.net.URI;
+
+/**
+ * GatewayInstanceInitializer takes care about starting the lookup for Gateway instance after the context is started
```

Comment:
"GatewayInstanceInitializer starts the lookup for Gateway instance after the context is started"
"takes care" is too vague